drafting.typ

```diff
@@ -192,53 +217,197 @@
+      panic(
+        "If the page width *and* height are set to `auto`, neither inside nor outside margin"
+          + " can be `auto`. Got (inside, outside) margin "
+          + repr((
+            margin.inside,
+            margin.outside,
+          ))
+          + ", and page (width, height) "
+          + repr((page.width, page.height)),
+      )
+    }
+  }
+  // https://github.com/typst/typst/issues/3636#issuecomment-1992541661
+  let page-dims = (page.width, page.height).filter(x => x != auto)
+  let auto-size = 2.5 / 21 * calc.min(..page-dims)
+
+  let defaults = (
+    margin-left: margin.left,
+    margin-right: margin.right,
+    margin-inside: margin.inside,
+    margin-outside: margin.outside,
+  )
+  for (k, v) in defaults {
+    if v == auto {
+      defaults.at(k) = auto-size
+    } else if type(v) == relative {
+      defaults.at(k) = v.length
+    } // ignore if none
+  }
+
+  return defaults
+}
+
 
 /// Required for `margin-note` to work, since it informs `drafting` of the page setup.
 #let set-page-properties(..kwargs) = {
   show: place // Avoid extra whitespace
   context {
     let kwargs = kwargs.named()
     layout(size => {
-      let margin = page.margin
-      if type(margin) != dictionary {
-        margin = (left: margin, right: margin)
-      }
-      if (auto in (margin.left, margin.right) and (page.width, page.height) == (auto, auto)) {
-        panic("If the page width *and* height are set to `auto`, neither left nor right margin" + " can be `auto`. Got (left, right) margin " + repr((
-          margin.left,
-          margin.right,
-        )) + ", and page (width, height) " + repr((page.width, page.height)))
-      }
-      // https://github.com/typst/typst/issues/3636#issuecomment-1992541661
-      let page-dims = (page.width, page.height).filter(x => x != auto)
-      let defaults = (
-        margin-left: margin.left,
-        margin-right: margin.right,
-      )
-      for (k, v) in defaults {
-        if v == auto {
-          defaults.at(k) = 2.5 / 21 * calc.min(..page-dims)
-        } else if type(v) == type(10pt - 100%) {
-          defaults.at(k) = v.length
-        }
+      let margins = _get-margins()
+
+      let binding = if page.binding == auto {
+        if text-direction(text.dir, text.lang) == ltr {
+          left
+        } else { right }
+      } else {
+        page.binding
       }
+
       set-margin-note-defaults(
-        ..defaults,
+        ..margins,
         page-width: size.width,
+        page-binding: binding,
         ..kwargs,
       )
     })
   }
 }
 
+#let _get-note-outline-props(note) = {
+  let func = note.func()
+  let defaults = margin-note-defaults.get()
+  // styled types from custom notes have unknown fill/stroke/body
+  // TODO: Maybe a better way to signify "unknown"?
+  let props = (
+    fill: note.at("fill", default: defaults.fill),
+    stroke: note.at("stroke", default: defaults.stroke),
+    body: note.at("body", default: "<body text unkonwn>"),
```

Comment:
Styled notes add things to the overview. I think it's fine for now, but we can think of a future PR to grab just the text
![image](https://github.com/user-attachments/assets/960ffcaa-93f6-4199-9cdf-058eaee92586)

Reply:
@Tinggaard Do you think it's desirable to copy the entire `stroke` instead of just the `paint` for the outline box, which will preserve dashing etc?

Reply:
Hmm, I'm hesitant to import another package for 1 function but it may be worthwhile. We can indeed fix it in a future PR if it's relevant.